lib/index.js

```diff
@@ -90,8 +90,43 @@ MapboxGeocoder.prototype = {
     }
   },
 
+  /**
+   * Add the geocoder to a container. The container can be either a mapboxgl.Map or a reference to an HTML class or ID. 
+   * 
+   * If the container is a mapboxgl.Map, this function will behave identically to Map.addControl(geocoder)
+   * If the container is an html id or class, the geocoder will be appended to that element
+   * 
+   * This function will throw an error if the container is not one a map or a class/id reference
+   * It will also throw an error if the referenced html element cannot be found in the document.body
+   * @param {String|mapboxgl.Map} container A reference to the container to which to add the geocoder 
+   */
+  addTo: function(container){
```

Comment:
Is this the new recommended way to invoke the geocoder? If so we should update the debug/index.js and docs to recommend using this method first, and not `onAdd` (which I think was always awkward so this is a welcome change)

Reply:
I think the recommended method should still be `map.addControl(geocoder)` -- since I believe this will still be the primary usage. We can add an example that uses this method though.

Reply:
Makes sense. So `map.addControl` if you have a map, otherwise use `addTo(container)`

This is the example I had in mind that we could update to use `addTo` instead of `onAdd` since its more logical: https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-outside-the-map/